shop-ui/src/main/java/com/maxshop/viewModel/CategoriesViewModel.kt

```diff
@@ -3,59 +3,63 @@ package com.maxshop.viewModel
 import androidx.lifecycle.MutableLiveData
 import com.example.internetshop.presentation.utils.SingleLiveEvent
 import com.maxshop.mapper.CategoryMapper
+import com.maxshop.model.RecyclerItem
 import com.maxshop.usecase.GetCategoriesUseCase
 import com.maxshop.viewState.CategoryViewState
 import io.reactivex.android.schedulers.AndroidSchedulers
 import io.reactivex.schedulers.Schedulers
-import java.util.concurrent.TimeUnit
 import javax.inject.Inject
 
 class CategoriesViewModel @Inject constructor(
     private val getCategoriesUseCase: GetCategoriesUseCase,
     private val categoryMapper: CategoryMapper
 ) : BaseViewModel() {
 
-    val events = SingleLiveEvent<CategoryEvents>()
-    val categoriesLiveData = MutableLiveData<List<CategoryViewState>>()
+    val events = SingleLiveEvent<Event>()
```

Comment:
It's not a good idea to have public mutable live data (SingleLiveEvent). If you mutable live data is public this means that someone can send event outside. This can lead to several errors and violates encapsulation principle.

To avoid this you need to create two variables : private mutable live data and public non mutable live data. Please do following:
private val _events = SingleLiveEvent<Event>() 
    val events :LiveData<Event> get() = _events
    
    Also, please make similar changes for the variable `categories`

Reply:
fixed